File: v3-sandbox/prototype-keys-java/src/main/java/org/hiero/keys/io/KeyEncoding.java
package org.hiero.keys.io;

import org.hiero.keys.KeyType;
import org.hiero.keys.impl.DerUtil;
import org.hiero.keys.impl.PemUtil;

/**
 * Supported encodings that can be used together with a container format.
 */
public enum KeyEncoding {
    DER(RawFormat.BYTES),
    PEM(RawFormat.STRING);

    private final RawFormat format;

    KeyEncoding(RawFormat format) {
        this.format = format;
    }

    public RawFormat getFormat() {
        return format;
    }

    public byte[] decode(final KeyType keyType, final String value) {
        return switch (this) {
            case DER -> DerUtil.decode(value);
            case PEM -> PemUtil.fromPem(keyType, value);
        };
    }
}

File: v3-sandbox/prototype-python/src/hiero_sdk_v3/keys/io/key_encoding.py
from enum import Enum
from .raw_format import RawFormat


class KeyEncoding(Enum):
    DER = "DER"
    PEM = "PEM"

    @property
    def raw_format(self) -> RawFormat:
        return RawFormat.BYTES if self == KeyEncoding.DER else RawFormat.STRING
